src/ProjectNeon/Assets/Features/Battle/Setup/SetupEnemies.cs

```diff
@@ -5,10 +5,20 @@ public class SetupEnemies : MonoBehaviour
     [SerializeField] private EnemyArea enemyArea;
     [SerializeField] private EncounterBuilder encounterBuilder;
     [SerializeField] private GameEvent onEncounterGenerated;
+    [SerializeField] private GameEvent onEnemyTurnsConfirmed;
```

Comment:
This is not part of the Setup flow.

Enemies must pick cards every turn. Setup happens only once per battle.